### PR TITLE
Fix/Textfield Auto Correct - (PBS-1599)

### PR DESCRIPTION
**Pull request by bmorris-revacomm**
- replace normal textfields with tabbed ones (where needed)
- refactor tabbed textfield
- fix textfields tab indices

### CODE REVIEW

File: PBLogging/Views/RecentSorties/RecentSortiesList.swift [hunk 1]
```diff
@@ -26,16 +26,18 @@ struct RecentSortiesList: View {
...
                         
-                        TextField("Event Summary", text: $mission.summary, onEditingChanged: { _ in
-                            dataController.save()
-                        })
-                        .withBorderLabel("EVENT SUMMARY")
+                        TabbedTextField(text: $mission.summary, tabIndex: 2001, placeholder: "Event Summary")
+                            .onChange(of: mission.summary) { _ in
+                                dataController.save()
+                            }
+                            .withBorderLabel("EVENT SUMMARY")
```
Comment (jbethanco) [inline PBLogging/Views/RecentSorties/RecentSortiesList.swift:40]: Default behavior of TabbedTextField keeps the keyboard in uppercase. Doesn't seem right for the Mission name/ summary fields.

File: PBLogging/Views/Overview/PasscodeView.swift [hunk 2]
```diff
@@ -32,17 +32,16 @@ struct PasscodeView: View {
             
             Group {
                 HStack {
-                    TextField("Unit name", text: $username)
-                    .onChange(of: username) { _ in checkPasscode() }
+                    TabbedTextField(text: $username, tabIndex: 1000, placeholder: "Unit name")
+                        .onChange(of: username) { _ in checkPasscode() }
```
Comment (jbethanco) [inline PBLogging/Views/Overview/PasscodeView.swift:36]: This defaults to uppercase too.
The unit name and passcode fields no longer have padding and are tight together.

File: PBLogging/Views/SingleSortie/AircrewListView.swift [hunk 3]
```diff
@@ -111,31 +111,26 @@ struct AircrewRow: View {
...
+                    .foregroundColor(crewLine.isAircraftCommander ? .pblForegroundPrimary : .pblForegroundSecondary)
+                    .opacity(determineImageOpacity())
+            }
+            .buttonStyle(PlainButtonStyle())
+            .disabled(!crewLine.isPilot)
+            
+            Group {
+                TabbedTextField(text: $crewLine.person.lastName, tabIndex: 5000, placeholder: "LAST NAME", autocapitalizationType: .words)
+                TabbedTextField(text: $crewLine.person.last4.onChange(enforce: .ssnFormatting), tabIndex: 5001, placeholder: "SSN", keyboardType: .numberPad)
+                TabbedTextField(text: $crewLine.flyingOrganization.onChange(enforce: .flyingOrgFormatting), tabIndex: 5002, placeholder: "0016", keyboardType: .numberPad)
```
Comment (jbethanco) [inline PBLogging/Views/SingleSortie/AircrewListView.swift:128]: Cool... didn't think it would work between rows, but the tabbing does.

**Review by jbethanco — CHANGES_REQUESTED** [review]
Overall looks good. I think we should take a hard look at which fields should default to caps.

File: PBLogging/Views/RecentSorties/RecentSortiesList.swift (hunk 1 above)
Comment (bmorris-revacomm) [reply PBLogging/Views/RecentSorties/RecentSortiesList.swift:40]: Oooooh good catch

**Review by jbethanco — APPROVED** [review]
Works great. Love that sentence vs word caps.

**Pull request merged**